File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
      int count = 0;

      int candidate = 0;

      foreach(int num in nums){
        if(count == 0){
          candidate = num; //if havent counted, then let candidate= num
        }
        if(num == candidate){
          count += 1;
        }
        else{
          count+= -1;
        }
      }
      return candidate;

        
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        standard = len(nums) // 2

        count = {}

        for num in nums:
          if num not in count:
            count[num] = 0
          count[num] += 1
          if count[num] > standard:
            return num